core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -736,30 +736,31 @@ private[spark] class Master(
     val appName = app.desc.name
     val notFoundBasePath = HistoryServer.UI_PATH_PREFIX + "/not-found"
     try {
-      val eventLogFile = app.desc.eventLogDir
-        .map { dir => EventLoggingListener.getLogPath(dir, app.id) }
+      val eventLogDir = app.desc.eventLogDir
         .getOrElse {
           // Event logging is not enabled for this application
           app.desc.appUiUrl = notFoundBasePath
           return false
         }
-        
-      val fs = Utils.getHadoopFileSystem(eventLogFile, hadoopConf)
-
-      if (fs.exists(new Path(eventLogFile + EventLoggingListener.IN_PROGRESS))) {
+      
+      val eventLogFilePrefix = EventLoggingListener.getLogPath(eventLogDir, app.id)        
+      val fs = Utils.getHadoopFileSystem(eventLogDir, hadoopConf)
+      val eventLogFileSuffix = if (fs.exists(new Path(eventLogFilePrefix + 
+          EventLoggingListener.IN_PROGRESS))) {
         // Event logging is enabled for this application, but the application is still in progress
-        val title = s"Application history not found (${app.id})"
-        var msg = s"Application $appName is still in progress."
+        var msg = s"Application $appName is still in progress, it may be terminated accidently."
```

Comment:
Hi @srowen , when app is running, it will recording the history at the same time. And the history log file name is with the format "app-xxxxx.inprogress" and after the app finished running, the suffix (.inprogress) is removed. And if app is stopped abnormally, the suffix remains.